ckan/lib/dictization/model_dictize.py

```diff
@@ -696,19 +634,29 @@ def vocabulary_list_dictize(vocabulary_list, context):
     return [vocabulary_dictize(vocabulary, context)
             for vocabulary in vocabulary_list]
 
-def activity_dictize(activity, context):
+def activity_dictize(activity, context, include_data=False):
     activity_dict = d.table_dictize(activity, context)
+    if not include_data:
+        # take a copy of the activity data, since the original may be used
+        # elsewhere during the same render and we don't want to affect that
+        activity_dict['data'] = copy.deepcopy(activity_dict['data'])
+        # delete all the data apart from the title field on each data object,
+        # because that is needed to display it in the activity stream
+        for obj_key in activity_dict['data'].keys():
+            obj_data = activity_dict['data'][obj_key]
+            if isinstance(obj_data, dict):
+                for key in obj_data.keys():
+                    if key != 'title':
+                        del obj_data[key]
+            else:
+                del activity_dict['data'][obj_key]
```

Comment:
You could replace the deepcopy and loops with
```python
activity_dict = {
    key: {'title': val['title']}
    for (key, val) in activity_dict.items() if 'title' in val}
```

Reply:
that is just the ticket! :) I've put that in now.